src/validators/beastValidators.ts

```diff
@@ -0,0 +1,69 @@
+import { body, oneOf, param } from 'express-validator';
+
+/**
+ * Input validation checks for creating a beast
+ * @returns {object} The validation checks
+ */
+export const validationForCreateBeast = () => [body('gamerTag')
+    .isString()
+    .withMessage('GamerTag must be a string')
+    .bail()
+    .notEmpty()
+    .withMessage('GamerTag is required')
+    .bail()
+    .isLength({ max: 20 })
+    .withMessage('GamerTag must be less than 20 characters')
+    .bail()
+    .matches(/^[a-zA-Z0-9_.]+$/)
+    .withMessage('GamerTag contains invalid characters'),
+body('email')
+    .isString()
+    .withMessage('Email must be a string')
+    .bail()
+    .notEmpty()
+    .withMessage('Email is required')
+    .bail()
+    .isEmail()
+    .withMessage('Email is not valid')];
+
+/**
+ * Input validation checks for updating a beast
+ */
+export const validationForUpdateBeast = () => [body('gamerTag')
+    .optional({ values: 'falsy' })
+    .isString()
+    .withMessage('GamerTag must be a string')
+    .bail()
+    .notEmpty()
+    .withMessage('GamerTag is required')
+    .bail()
+    .isLength({ max: 20 })
+    .withMessage('GamerTag must be less than 20 characters')
+    .bail()
+    .matches(/^[a-zA-Z0-9_.]+$/)
+    .withMessage('GamerTag contains invalid characters'),
+body('email')
+    .optional({ values: 'falsy' })
```

Comment:
Why is values falsy here rather than undefined/null?

Reply:
falsy is the one that is supposed to be used for optional form input:

Here are examples

1. .optional({ values: 'falsy' })

This setting considers any falsy value as optional, meaning the validation chain will not trigger if the field contains any of the following falsy values:


Input: { gamerTag: "" } — Does not trigger validation.
Input: { gamerTag: 0 } — Does not trigger validation.
Input: { gamerTag: false } — Does not trigger validation.
Input: { gamerTag: null } — Does not trigger validation.
Input: { gamerTag: undefined } — Does not trigger validation.
Input: { } (field not provided) — Does not trigger validation.
Input: { gamerTag: "Player1" } — Triggers validation.

2. .optional({ values: 'undefined' })
This setting only considers undefined as optional, which means the field is optional only if it is completely omitted or explicitly set to undefined:

Input: { gamerTag: "" } — Triggers validation.
Input: { gamerTag: 0 } — Triggers validation.
Input: { gamerTag: false } — Triggers validation.
Input: { gamerTag: null } — Triggers validation.
Input: { gamerTag: undefined } — Does not trigger validation.
Input: { } (field not provided) — Does not trigger validation.
Input: { gamerTag: "Player1" } — Triggers validation.

3. .optional({ values: 'null' })
This setting considers both undefined and null as optional. The field is considered optional if it is not provided, set to null, or set to undefined:


Input: { gamerTag: "" } — Triggers validation.
Input: { gamerTag: 0 } — Triggers validation.
Input: { gamerTag: false } — Triggers validation.
Input: { gamerTag: null } — Does not trigger validation.
Input: { gamerTag: undefined } — Does not trigger validation.
Input: { } (field not provided) — Does not trigger validation.
Input: { gamerTag: "Player1" } — Triggers validation.